### PR TITLE
Finished topic 1

### PR DESCRIPTION
**Pull request by macasaenz**

### CODE REVIEW

File: maca/00-HTML-CSS-basics/index.html [hunk 1]
```diff
@@ -10,25 +10,30 @@
...
+		<aside class="sidebar"> <h3> This is the ASIDE section </h3> 
+			<p class="oh-no" style="background:red">
+				Testing CSS specificity 
+			</p>
+		</aside>
+
 	</div>
 
-	<footer id="mainFooter"> MAIN FOOTER </footer>
+	<footer id="mainFooter" style="color: blue" class="footer"> MAIN FOOTER </footer>
```
Comment (marcelososa) [inline maca/00-HTML-CSS-basics/index.html]: Avoid inline styles.

File: maca/00-HTML-CSS-basics/styles/style.css [hunk 2]
```diff
@@ -55,25 +56,26 @@
...
+  font-weight: bolder;
 }
 
-#mainFooter{
+.footer{
+  font-size: 10px;
   margin-top: 10px;
   background: darkgreen;
-  color: white;
+  color: white !important;
```
Comment (marcelososa) [inline maca/00-HTML-CSS-basics/styles/style.css]: Why use `!important!`?

File: maca/00-HTML-CSS-basics/styles/style.css [hunk 3]
```diff
@@ -85,3 +87,20 @@ header, footer {
...
+[class*="a"]:not([class*="r"]){
+  background: blue;
+}
+
+[class$="r"]{
+  background: magenta;
+}
+
+.oh-no{
+  background: green !important;
```
Comment (marcelososa) [inline maca/00-HTML-CSS-basics/styles/style.css]: Why use `!important!`?

File: maca/01-JS-DOM-APIs/assets/js/index.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
+let config = {
+					"method": "",
+					"url": "",
+					"q": "", 
+				}
+
+
+
+window.onload = function(){			
+			document.getElementById('hidden').style.opacity = 1;
```
Comment (marcelososa) [inline maca/01-JS-DOM-APIs/assets/js/index.js]: Styles should be handled by CSS classes.

File: maca/01-JS-DOM-APIs/assets/js/index.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+window.onload = function(){			
+			document.getElementById('hidden').style.opacity = 1;
+			document.getElementById("btn").addEventListener("click", getJoke2);
+			document.getElementById("btn2").addEventListener("click", getNames);
+
+}
+
+//topic 1 - ex. 6
+getJoke = function(){
```
Comment (marcelososa) [inline maca/01-JS-DOM-APIs/assets/js/index.js]: The right way to declare a function is:
```javascript
function getJoke(){};
```
or 
```javascript
var getJoke = function(){};
```
or the best:
```javascript
var getJoke = function getJoke(){};
```

File: maca/01-JS-DOM-APIs/assets/js/index.js [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+};
+
+//topic 1 - ex. 9
+getNames = function(){	
+	config.method = "GET";
+	config.q = document.getElementById('q').value;
+	config.url = "https://api.github.com/search/repositories?q="+config.q;
+	promiseFunct(config)
+	.then(function(response){
+		list = document.createElement("ul");
```
Comment (marcelososa) [inline maca/01-JS-DOM-APIs/assets/js/index.js]: This variable is undeclared, at this point is stored and visible at global level.

File: maca/01-JS-DOM-APIs/assets/styles/main.css [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+  margin-left: 1%;
+  margin-right: 1%; 
+}
+
+header{
+  text-align: center;
+  background: #99ccff;
+}
+
+#hidden{
```
Comment (marcelososa) [inline maca/01-JS-DOM-APIs/assets/styles/main.css]: IDs should be unique and specific, this is very generic.

File: maca/01-JS-DOM-APIs/assets/styles/main.css [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+  text-align: center;
+  background: #99ccff;
+}
+
+.jokeWrap{
+  border-color: #33bbff;
+  border-style: dashed;
+}
+
+#btn{
```
Comment (marcelososa) [inline maca/01-JS-DOM-APIs/assets/styles/main.css]: IDs should be unique and specific, this is very generic.

File: maca/01-JS-DOM-APIs/assets/styles/main.css [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+  display: table;
+  margin-top: 2%;
+  margin-left: auto;
+  margin-right: auto; 
+ }
+
+.jokeOK{
+  background: #66ffcc !important;
```
Comment (marcelososa) [inline maca/01-JS-DOM-APIs/assets/styles/main.css]: Why use `!important!`?

File: maca/01-JS-DOM-APIs/assets/styles/main.css [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+.jokeError{
+  background: red !important;
```
Comment (marcelososa) [inline maca/01-JS-DOM-APIs/assets/styles/main.css]: Why use `!important!`?

File: maca/01-JS-DOM-APIs/assets/styles/main.css [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+#btn2{
```
Comment (marcelososa) [inline maca/01-JS-DOM-APIs/assets/styles/main.css]: IDs should be unique and specific, this is very generic.

File: maca/01-JS-DOM-APIs/assets/styles/main.css [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+    margin-top: 5%;
+    margin-left: 2%;
+}
+
+#q{
```
Comment (marcelososa) [inline maca/01-JS-DOM-APIs/assets/styles/main.css]: IDs should be unique and specific, this is very generic.

File: maca/00-HTML-CSS-basics/index.html (hunk 1 above)
Comment (macasaenz) [inline maca/00-HTML-CSS-basics/index.html]: Done

File: maca/00-HTML-CSS-basics/styles/style.css (hunk 2 above)
Comment (macasaenz) [inline maca/00-HTML-CSS-basics/styles/style.css]: Done, i must have been trying something and forgot to delete it.

File: maca/00-HTML-CSS-basics/styles/style.css (hunk 3 above)
Comment (macasaenz) [inline maca/00-HTML-CSS-basics/styles/style.css]: I thought this was the solution to ex. 2.3 7: 

Imagine there is a declaration like class=”oh-no-inline-styles” style=”background:red” and you need to change the background to green without changing the inline style. How could you accomplish this?

?

File: maca/01-JS-DOM-APIs/assets/js/index.js (hunk 5 above)
Comment (macasaenz) [inline maca/01-JS-DOM-APIs/assets/js/index.js]: Done

File: maca/01-JS-DOM-APIs/assets/js/index.js (hunk 6 above)
Comment (macasaenz) [inline maca/01-JS-DOM-APIs/assets/js/index.js]: Done

File: maca/01-JS-DOM-APIs/assets/styles/main.css (hunk 8 above)
Comment (macasaenz) [inline maca/01-JS-DOM-APIs/assets/styles/main.css]: Done

File: maca/01-JS-DOM-APIs/assets/styles/main.css (hunk 11 above)
Comment (macasaenz) [inline maca/01-JS-DOM-APIs/assets/styles/main.css]: Done

File: maca/01-JS-DOM-APIs/assets/styles/main.css (hunk 12 above)
Comment (macasaenz) [inline maca/01-JS-DOM-APIs/assets/styles/main.css]: Done

File: maca/01-JS-DOM-APIs/assets/styles/main.css (hunk 7 above)
Comment (macasaenz) [inline maca/01-JS-DOM-APIs/assets/styles/main.css]: Done

File: maca/01-JS-DOM-APIs/assets/js/index.js (hunk 4 above)
Comment (macasaenz) [inline maca/01-JS-DOM-APIs/assets/js/index.js]: Done

File: maca/01-JS-DOM-APIs/assets/styles/main.css (hunk 9 above)
Comment (macasaenz) [inline maca/01-JS-DOM-APIs/assets/styles/main.css]: I needed it to overrite the background property set by id, but now i changed it to use classes so i dont need it anymore. 
Done

File: maca/01-JS-DOM-APIs/assets/styles/main.css (hunk 10 above)
Comment (macasaenz) [inline maca/01-JS-DOM-APIs/assets/styles/main.css]: I needed it to overrite the background property set by id, but now i changed it to use classes so i dont need it anymore. 
Done

File: maca/00-HTML-CSS-basics/styles/style.css (hunk 3 above)
Comment (marcelososa) [inline maca/00-HTML-CSS-basics/styles/style.css]: Ohhh I saw, you're right!
Forget this comment.

Forget thid

**Comment by marcelososa** [comment]
LGTM 👍

**Pull request merged**